dipy/reconst/dki_micro.py

```diff
@@ -0,0 +1,507 @@
+            3) Fifteen elements of the kurtosis tensor
+    sphere : Sphere class instance, optional
+        The sphere providing sample directions for the initial search of the
+        maximal value of kurtosis.
+    gtol : float, optional
+        This input is to refine kurtosis maxima under the precision of the
+        directions sampled on the sphere class instance. The gradient of the
+        convergence procedure must be less than gtol before successful
+        termination. If gtol is None, fiber direction is directly taken from
+        the initial sampled directions of the given sphere object
+    mask : ndarray
+        A boolean array used to mark the coordinates in the data that should be
+        analyzed that has the shape dki_params.shape[:-1]
+
+    Returns
+    --------
+    AWF : ndarray (x, y, z) or (n)
+        Axonal Water Fraction
+
+    References
+    ----------
+    .. [1] Fieremans E, Jensen JH, Helpern JA, 2011. White matter
+           characterization with diffusional kurtosis imaging.
+           Neuroimage 58(1):177-88. doi: 10.1016/j.neuroimage.2011.06.006
+    """
+    kt_max = kurtosis_maximum(dki_params, sphere=sphere, gtol=gtol, mask=mask)
+
+    awf = kt_max / (kt_max + 3)
+
+    return awf
+
+
+def diffusion_components(dki_params, sphere='repulsion100', awf=None,
+                         mask=None):
+    """ Extracts the restricted and hindered diffusion tensors of well aligned
+    fibers from diffusion kurtosis imaging parameters [1]_.
+
+    Parameters
+    ----------
+    dki_params : ndarray (x, y, z, 27) or (n, 27)
+        All parameters estimated from the diffusion kurtosis model.
+        Parameters are ordered as follows:
+            1) Three diffusion tensor's eigenvalues
+            2) Three lines of the eigenvector matrix each containing the first,
+               second and third coordinates of the eigenvector
+            3) Fifteen elements of the kurtosis tensor
+    sphere : Sphere class instance, optional
+        The sphere providing sample directions to sample the restricted and
+        hindered cellular diffusion tensors. For more details see Fieremans
+        et al., 2011.
+    awf : ndarray (optional)
+        Array containing values of the axonal water fraction that has the shape
+        dki_params.shape[:-1]. If not given this will be automatically computed
+        using :func:`axonal_water_fraction`" with function's default precision.
+    mask : ndarray (optional)
+        A boolean array used to mark the coordinates in the data that should be
+        analyzed that has the shape dki_params.shape[:-1]
+
+    Returns
+    --------
+    edt : ndarray (x, y, z, 6) or (n, 6)
+        Parameters of the hindered diffusion tensor.
+    idt : ndarray (x, y, z, 6) or (n, 6)
+        Parameters of the restricted diffusion tensor.
+
+    Notes
+    -----
+    The parameters of both hindered and restricted diffusion tensors are order
```

Comment:
I think this note is inconsistent with the rest of the docstring. It seems to suggest that we return these tensors as 12 parameters each (three eigenvalues + three eigenvectors).